### PR TITLE
Added intercepted channel registry in kafka streams app for its implementers

### PR DESCRIPTION
**Pull request by singhalprerana**

### CODE REVIEW

File: kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/KafkaStreamsApp.java [hunk 1]
```diff
@@ -58,6 +63,9 @@ public abstract class KafkaStreamsApp extends PlatformService {
   public static final String PRE_CREATE_TOPICS = "precreate.topics";
   public static final String KAFKA_STREAMS_CONFIG_KEY = "kafka.streams.config";
   private static final Logger logger = LoggerFactory.getLogger(KafkaStreamsApp.class);
+
+  protected final GrpcChannelRegistry grpcChannelRegistry;
```
Comment (aaron-steinfeld) [inline kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/KafkaStreamsApp.java:67]: could you put this behind a protected getter?

**Review by aaron-steinfeld — COMMENTED** [review]
Otherwise, lgtm

File: kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/KafkaStreamsApp.java [hunk 2]
```diff
@@ -67,6 +75,13 @@ public abstract class KafkaStreamsApp extends PlatformService {
 
   public KafkaStreamsApp(ConfigClient configClient) {
     super(configClient);
+    this.grpcChannelRegistry =
```
Comment (kotharironak) [inline kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/KafkaStreamsApp.java:78]: Do all the kstream apps need this? Shouldn't this be an extension of GrpcKstreamApp or something?

Comment (aaron-steinfeld) [reply kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/KafkaStreamsApp.java:78]: > Do all the kstream apps need this?

Yes. Almost all of them use it for their application logic anyway, but all of them will need to use grpc to use the weighted group partitioner.

File: kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/KafkaStreamsApp.java (hunk 1 above)
Comment (singhalprerana) [reply kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/KafkaStreamsApp.java]: done

**Review by aaron-steinfeld — APPROVED** [review]

**Pull request merged**